apps/mobile/src/features/token/use-get-token-details.ts

```diff
@@ -0,0 +1,28 @@
+import { toFetchState } from '@/components/loading';
+import {
+  useAssetDescriptionQuery,
+  useAssetPriceChangeQuery,
+} from '@/queries/assets/fungible-asset-info.query';
+import { useMarketDataQuery } from '@/queries/market-data/market-data.query';
+
+import { FungibleCryptoAsset } from '@leather.io/models';
+
+export function useGetTokenDetails({ asset }: { asset: FungibleCryptoAsset }) {
```

Comment:
This hook groups the main token-details queries together and adds `fetchState` to help with more elegant loading. 

For SIP-10 - the `asset` needs to come from the `balance` query which sometimes hasn't completed when the user lands on the page.